nebula-logger/core/main/configuration/classes/LoggerCache.cls

```diff
@@ -0,0 +1,230 @@
+//------------------------------------------------------------------------------------------------//
+// This file is part of the Nebula Logger project, released under the MIT License.                //
+// See LICENSE file or go to https://github.com/jongpie/NebulaLogger for full license details.    //
+//------------------------------------------------------------------------------------------------//
+
+/**
+ * @group Configuration
+ * @description Class used to cache query results returned by the selector classes
+ */
+public without sharing class LoggerCache {
+    private static final Boolean CACHE_IS_IMMUTABLE = false;
+    @TestVisible
+    private static final String CACHE_PARTITION_NAME = getQualifiedParitionName('LoggerCache');
+    private static final Cache.Visibility CACHE_VISIBILITY = Cache.Visibility.NAMESPACE;
+    private static final Cache.OrgPartition ORGANIZATION_CACHE_PARTITION = Cache.Org.getPartition(CACHE_PARTITION_NAME);
+    private static final Integer ORGANIZATION_CACHE_TTL_SECONDS = 86400; // 86,400 seconds == 24 hours, max allowd for org cache
+    private static final Cache.SessionPartition SESSION_CACHE_PARTITION = Cache.Session.getPartition(CACHE_PARTITION_NAME);
+    private static final Integer SESSION_CACHE_TTL_SECONDS = 28800; // 28,800 seconds == 8 hours, max allowd for session cache
+
+    private static final OrganizationCache ORGANIZATION_CACHE_INSTANCE = new OrganizationCache();
+    private static final SessionCache SESSION_CACHE_INSTANCE = new SessionCache();
+    private static final TransactionCache TRANSACTION_CACHE_INSTANCE = new TransactionCache();
+
+    @SuppressWarnings('PMD.ApexDoc')
+    private interface Cacheable {
```

Comment:
@jamessimone I meant to ask you about this earlier. Right now, the `Cacheable` interface is private, and each of the static methods returns a concrete class that implements the interface (and the 3 concrete inner classes are public - `OrganizationCache`, `SessionCache` & `TransactionCache`).

I'm thinking of instead making the `Cacheable` interface public, updating the static methods' returns types to `Cacheable` (instead of the concrete classes), and then make the 3 concrete inner classes `private`. Thoughts on this potential change? I conceptually like the idea because it would completely abstract away the internal implementations - external code would only know about the interface.

Reply:
That would be a great change!

Reply:
Alright, then I'm going for it!

Reply:
Done!